### ISSUE TITLE
Get currently selected artwork

### ISSUE THREAD

**Issue by andrasferenczi** [romannurik/muzei#689]
This is what I am trying to do:

> Add new artworks to Muzei. Once there are too many artworks, delete the unused ones, unless the user has that one selected.

Implementation works with adding and removing artwork, but I cannot understand how to retrieve the current artwork.

So I know about the `MuzeiContract.Artwork.getCurrentArtwork` function, the issue is that this artwork does not have an ID and  it has a different created date than the ones retrieved using the cursor.

I also don't understand why the implementation of `getCurrentArtwork` works:

```kotlin
        @JvmStatic
        fun getCurrentArtwork(context: Context): com.google.android.apps.muzei.api.Artwork? {
            val contentResolver = context.contentResolver
            return contentResolver.query(CONTENT_URI, null, null, null, null)?.use { cursor ->
                if (cursor.moveToFirst()) {
                    com.google.android.apps.muzei.api.Artwork.fromCursor(cursor)
                } else {
                    null
                }
            }
        }
```

It looks like this code just accesses the first one and I don't see any guarantee why that one is the selected.

What is not clear to me, is that this `readArtworks` extension (custom) function works and retrieves `com.google.android.apps.muzei.api.provider.Artwork` (the one with id), but it's first one is not the selected one.

```kotlin
fun ProviderClient.readArtworks(context: Context): List<Artwork> {
    return context.contentResolver
        .query(contentUri, null, null, null, null)
        .use { cursor ->

            if (cursor == null) {
                return@use emptyList<Artwork>()
            }

            val result = mutableListOf<Artwork>()

            cursor.moveToFirst()

            while (!cursor.isAfterLast) {
                val artwork = Artwork.fromCursor(cursor)
                result.add(artwork)
                cursor.moveToNext()
            }

            return@use result
        }
}
```
---

In short, I need to convert `com.google.android.apps.muzei.api.Artwork` somehow into `com.google.android.apps.muzei.api.provider.Artwork`.

It seems to me that adding a `UUID` hash as it's `metadata` would solve this issue (as I can find the corresponding artwork), but this feels hacky to me and I haven't even fully grasped why there are 2 different classes.

(If `metadata` does not work, I could still compare `byLine`, `title` and `attribution`, which would work in most cases probably.)

Can I please get a hint on this one?

**Comment by ianhanniballake** [comment]
Your base premise:

> Add new artworks to Muzei. Once there are too many artworks, delete the unused ones, unless the user has that one selected.

Is not something you should be doing. It is totally and 100% okay for you to have 10,000+ artworks in your `MuzeiArtProvider`. Muzei will ensure only the last ~50 artwork are cached. If you are having problems with those cases, please file those as separate issues and we can take a look.

That being said, you'll note that the two cases you're looking at are talking to entirely separate databases. The `MuzeiContract` talks to **Muzei's** database as per [the documentation](http://api.muzei.co/reference/com.google.android.apps.muzei.api/-muzei-contract/-artwork/index.html). Muzei's database adds a new row whenever any image is shown the user (if you were to swap between sources, you'd note that a new row would be added each time). This means that the first row is always the currently display image and why there is a separate `Artwork` class for that database - it is entirely independent of **your** `MuzeiArtProvider` and its database. That's also why your `readArtworks`, which reads from your `MuzeiArtProvider`, gets a very different result.

If you're insistent on going down the wrong direction of deleting artwork because you think there's too many artworks, you'd want to look at the [`imageUri`](http://api.muzei.co/reference/com.google.android.apps.muzei.api/-artwork/image-uri.html) of the returned `com.google.android.apps.muzei.api.Artwork` - you'll note that `Uri`s returned there are the URIs for a individual artwork from your `MuzeiArtProvider` (i.e., if you do `ContentUris.withAppendedId(providerClient.contentUri, artwork.id)`, you'll get the same Uri back).

**Comment by andrasferenczi** [comment]
Thank you for your fast response and for clearing it up. I didn't understand the separate databases, now I think I understand it a bit more and it makes much more sense now. In the end I went with the intended solution that fits more to `Muzei` by never removing images automatically.

Also, I added a `RemoteAction` to be able to remove a specific `Artwork`, so the user still has the option to clean up unwanted images from their gallery.

Thanks again. Great app and great support.
